src/components/CorrectedAgeCalculator.jsx

```diff
@@ -43,58 +43,72 @@ export default function CorrectedAgeCalculator() {
 
     // Calculate differences in years, months, weeks, and days
     let years = differenceInYears(today, birth);
-    const birthYear = birth.getFullYear();
-    const birthMonth = birth.getMonth();
-    let months = differenceInMonths(
-      today,
-      new Date(birthYear, birthMonth + years * 12)
-    );
-    let weeks = differenceInWeeks(
-      today,
-      new Date(birthYear, birthMonth + years * 12 + months, birth.getDate())
-    );
-    let days = differenceInDays(
-      today,
-      new Date(
-        birthYear,
-        birthMonth + years * 12 + months,
-        birth.getDate() + weeks * 7
-      )
-    );
-
+    let weeks;
+    // const birthYear = birth.getFullYear();
+    // const birthMonth = birth.getMonth();
+    // let months = differenceInMonths(
+    //   today,
+    //   new Date(birthYear, birthMonth + years * 12)
+    // );
+    // let weeks = differenceInWeeks(
+    //   today,
+    //   new Date(birthYear, birthMonth + years * 12 + months, birth.getDate())
+    // );
+    // let days = differenceInDays(
+    //   today,
+    //   new Date(
+    //     birthYear,
+    //     birthMonth + years * 12 + months,
+    //     birth.getDate() + weeks * 7
+    //   )
+    let days = differenceInDays(today,birth);
+
+    if (gestationWeeks > 40) gestationWeeks = 40; // [YH] as for birth after week 40, no need to fix the child age
     const prematureWeeks = FULL_TERM_WEEKS - gestationWeeks;
-
+    // [YH] the below updated calculation for 'years' is in order to consider the premature weeks
+    // [YH] so if gestationWeeks = 38, the baby will be considered 2 years only after 2 years and 2 weeks
+    years = differenceInYears(today, birth.setDate(birth.getDate() + (prematureWeeks * 7)));
+    
     if (years < 2) {
-      const ageInDays = years * 365 + months * 30 + weeks * 7 + days;
+      // const ageInDays = years * 365 + months * 30 + weeks * 7 + days;
+      // [YH] I replaced the above calculation, from 2 reasons: A) month=30days is wrong assumption. B) as only weeks are required, there is no need to calculate the months 
```

Comment:
If you want to add relevant correspondence about your changes you can add them like I add these comments inside the pr and not as part of the code.